apps/notification-producer/src/SubscriptionsRepository.ts

```diff
@@ -0,0 +1,21 @@
+import { getAllSubscribedAccounts } from '@cowprotocol/cms-api';
+
+const CACHE_TIME = 30000;
+
+// TODO: Move to repositories and make a proper cached repository and use DI
+export class SubscriptionRepository {
+  private lastCheck: number | null = null;
+  private cachedAccounts: string[] | null = null;
+
+  async getAllSubscribedAccounts(): Promise<string[]> {
+    const now = Date.now();
+    if (!this.lastCheck || now - this.lastCheck > CACHE_TIME) {
```

Comment:
Just to be more safe, we can check the condition to:

```
if (!this.cachedAccounts || !this.lastCheck || now - this.lastCheck > CACHE_TIME)
```

Otherwise, in theory we could return an empty array when there is a valid `lastCheck` but no `cachedAccounts`.

Reply:
I think it's fine because it could have cached no accounts.